src/Core/Framework/Adapter/Twig/TemplateFinder.php

```diff
@@ -72,15 +72,6 @@ public function find(string $template, $ignoreMissing = false, ?string $source =
 
         $queue = $this->getNamespaceHierarchy();
 
-        if ($source) {
```

Comment:
The array_merge here was incorrect. It completely reordered the queue instead of ensuring that the `$source` template was at the end.

A correct version would be:
```php
        if ($source) {
            $index = array_search($source, $queue, true);

            $queue = array_merge(
                array_slice($queue, 0, $index),
                array_slice($queue, $index + 1),
                [$source]
            );
        }
```

However, below in the foreach loop, you actually already do:
```
            if (!$this->loader->exists($name)) {
                continue;
            }
```

and after the foreach:
```php
            // if no other bundle extends the requested template, load the original template
            if (!$this->loader->exists($name)) {
                continue;
            }
```

and at the beginning of the loop there is actually a comment regarding this:
```
  // iterate over all bundles but exclude the originally requested bundle
        // example: if @Storefront/storefront/index.html.twig is requested, all bundles except Storefront will be checked first	        // example: if @Storefront/storefront/index.html.twig is requested, all bundles except Storefront will be checked first
```